File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/marketing/cs/distributecampaignfrommarketinglist4.cs
// <snippetdistributecampaignfrommarketinglist4>


 // Distribute and execute the campaign activity.
 // PostWorkflowEvent signals Microsoft Dynamics CRM to actually create the phone call activities.
 // Propagate also signals to Microsoft Dynamics CRM to create the phone call activities.
 // OwnershipOptions indicates whom the created activities should be assigned
 // to.
 var distributeRequest = new DistributeCampaignActivityRequest
 {
     Activity = phonecall,
     CampaignActivityId = _campaignActivityId,
     Owner = new EntityReference(
         SystemUser.EntityLogicalName, currentUser.UserId),
     OwnershipOptions = PropagationOwnershipOptions.Caller,
     PostWorkflowEvent = true,
     Propagate = true,
     SendEmail = false,
 };

 var distributeResponse = 
     (DistributeCampaignActivityResponse)_serviceProxy.Execute(distributeRequest);

 Console.WriteLine("  Distributed and executed the campaign activity to the marketing list.");

// </snippetdistributecampaignfrommarketinglist4>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/marketingvb/vb/distributecampaignfrommarketinglist4.vb
' <snippetdistributecampaignfrommarketinglist4>


	' Distribute and execute the campaign activity.
          ' PostWorkflowEvent signals Microsoft Dynamics CRM to actually create the phone call activities.
          ' Propagate also signals to Microsoft Dynamics CRM to create the phone call activities.
	' OwnershipOptions indicates whom the created activities should be assigned
	' to.
          Dim distributeRequest =
              New DistributeCampaignActivityRequest With
              {
                  .Activity = phonecall,
                  .CampaignActivityId = _campaignActivityId,
                  .Owner = New EntityReference(SystemUser.EntityLogicalName,
                                               currentUser.UserId),
                  .OwnershipOptions = PropagationOwnershipOptions.Caller,
                  .PostWorkflowEvent = True,
                  .Propagate = True,
                  .SendEmail = False
              }

          Dim distributeResponse =
              CType(_serviceProxy.Execute(distributeRequest), 
                  DistributeCampaignActivityResponse)

          Console.WriteLine(
              "  Distributed and executed the campaign activity to the marketing list.")

' </snippetdistributecampaignfrommarketinglist4>